src/parser/handle_heredoc.c

```
@@ -0,0 +1,119 @@
+	i = 0;
+	while (paths[i])
+	{
+		path = ft_xstrjoin(paths[i], HEREDOC_TMPFILE);
+		fd = open(path, O_WRONLY | O_CREAT | O_EXCL, S_IRUSR | S_IWUSR);
+		ft_strlcpy(tmpfile, path, HEREDOC_TMPFILE_LEN);
+		free(path);
+		if (fd != -1)
+			break ;
+		i++;
+	}
+	return (fd);
```

cppcheck

[warning] Uninitialized variable

(uninitvar)